### DISCUSSION TITLE
Custom Module: Customizing the Sidebar with v-sidebar-detail

### DISCUSSION THREAD

**Discussion by shadowcpy** [directus/directus#10609] [Archive]
Hey everyone!
I am currently trying to create a new module and make use of the sidebar slot in private-view.
But now I have hit a roadblock: How can I get access to the same fancy UI element (`v-sidebar-detail`) that is used throughout the app to display sidebar items? Whenever I use the other components, it looks fine as long as the drawer is open, however when it is closed, everything gets squished somewhat...
Has anyone thought of an idea to customize your own sidebar? Is there a way I could get access to `v-sidebar-detail` in my code and use it in the module?
Thanks for your help.

![image](https://user-images.githubusercontent.com/65440078/146676311-b718440f-6b4d-4db3-a76a-81a1d1ce4182.png)

![image](https://user-images.githubusercontent.com/65440078/146676323-48929ca7-9215-462a-a078-da98b1cb21ce.png)

![image](https://user-images.githubusercontent.com/65440078/146676337-dedc1d99-e8ea-41b8-ae04-f84ec5ebba1e.png)

![image](https://user-images.githubusercontent.com/65440078/146676332-ef44f90a-8862-454e-97aa-797288ee68a4.png)

**Comment by benhaynes** [comment]
@directus/team — We need a full (end to end) example of each extension type... both an article and a full example codebase/repo to work with.
